docs/source/audio_dataset_repo.mdx

```diff
@@ -0,0 +1,528 @@
+    """Yields examples as (key, example) tuples."""
+    examples = {}
+    with open(prompts_path, encoding="utf-8") as f:
+        for row in f:
+            data = row.strip().split(" ", 1)
+            speaker_id = data[0].split("_")[0]
+            audio_path = "/".join([path_to_clips, speaker_id, data[0] + ".wav"])
+            examples[audio_path] = {
+                "speaker_id": speaker_id,
+                "path": audio_path,
+                "sentence": data[1],
+            }
+    inside_clips_dir = False
+    id_ = 0
+    for path, f in audio_files:
+        if path.startswith(path_to_clips):
+            inside_clips_dir = True
+            if path in examples:
+                audio = {"path": path, "bytes": f.read()}
+                yield id_, {**examples[path], "audio": audio}
+                id_ += 1
+        elif inside_clips_dir:
+            break
+```
+
+### Upload the dataset to the Hub
+
+Once your script is ready, [create a dataset card](./dataset_card) and [upload it to the Hub](./share).
+
+Congratulations, you can now load your dataset from the Hub! 🥳
+
+```py
+>>> from datasets import load_dataset
+>>> load_dataset("<username>/my_dataset")
+```
+
+## (Advanced+) Extract TAR archives inside a script
+
+In the example above downloaded archives are not extracted and therefore examples do not contain information about where they are stored locally.
+To explain how to do the ectraction, we will briefly go through a [LibriVox Indonesia](https://huggingface.co/datasets/indonesian-nlp/librivox-indonesia) loading script.
+
+
+### Download and define the dataset splits
+
+1. Use the [`~DownloadManager.download`] method to download the audio data specified in `_AUDIO_URL`.
+
+2. To extract audio TAR archive locally, use the [`~DownloadManager.extract`]. You can use this method only in non-streaming mode (when `dl_manager.is_streaming=False`). This returns a local path to the extracted archive directory:
+
+   ```py
+   local_extracted_archive = dl_manager.extract(audio_path) if not dl_manager.is_streaming else None
+   ```
+
+3. Use the [`~DownloadManager.iter_archive`] method to iterate over the archive at `audio_path` after it's downloaded, just as in the Vivos example above. [`~DownloadManager.iter_archive`] doesn't provide any information about the full local paths of files inside the archive, even if they exist locally. As a result, you need to pass the `local_extracted_archive` path to the next step inside `gen_kwargs`, in order to preserve information about where the locally extracted files are. This is required to construct the correct paths to the local files when you generate the examples.
+
+<Tip>
+
+The reason you need to use a combination of [`~DownloadManager.download`] and [`~DownloadManager.iter_archive`] is because data in TAR archives can't be accessed directly from their paths. Instead, you'll need to download it first and then sequentially iterate over the files within the archive!
```

Comment:
(nit)
```suggestion
The reason you need to use a combination of [`~DownloadManager.download`] and [`~DownloadManager.iter_archive`] is because data in TAR archives can't be accessed directly from their paths inside the archive. A TAR archive is just a concatenation of files, so you'll need to download it first and then sequentially iterate over the files within the archive to find the one you want!
```